modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/ButtonTag.java

```diff
@@ -254,6 +255,32 @@ protected void cleanUp() {
 		_type = "button";
 	}
 
+	@Override
+	protected String getHydratedModuleName() {
+		if (Validator.isNotNull(getAdditionalProps()) ||
+			Validator.isNotNull(getPropsTransformer())) {
```

Comment:
So... this is a bit weird... passing `additionalProps` only really makes sense if you have a `propsTransformer`... otherwise, all props that matter are already available through tag attributes and passing additional ones won't matter unless you hydrate the component (and potentially pass them through a transformer).

The other case, providing only a transformer but no `additionalProps` could make sense for simple `onClick` methods or to do any kind of transformation over the original set of props. Changing props that affect the render of the component will cause FOUC (or whatever the name for this is in this context).